addon/ext.js

```diff
@@ -85,6 +85,11 @@ assign(RecordDataPrototype, {
   },
 
   didCommit(data) {
+    if (this._attributes) {
```

Comment:
I'm going to test this fix out in our app but should we be somehow making `willCommit` fire as that is the root cause?

Reply:
So I had a look into whether we could make `willCommit` fire as expected and it's a bit all over the shop.

The [`willCommit`](https://github.com/adopted-ember-addons/ember-data-model-fragments/blob/1403665eb8b8a3e43c31a8adccc1d301bed26d8f/addon/record-data.js#L281) hook here works by iterating over any fragments registered on the record by looking at the `this.fragments` class field. Unfortunately it appears that we only add fragments to `this.fragments` in cases where we specifically [`set` a fragment/fragment array](https://github.com/adopted-ember-addons/ember-data-model-fragments/blob/1403665eb8b8a3e43c31a8adccc1d301bed26d8f/addon/record-data.js#L164), but not when we initialise them.

I did a small spike where I also push fragments/fragment arrays into `this.fragments` in [`setupFragment`](https://github.com/adopted-ember-addons/ember-data-model-fragments/blob/1403665eb8b8a3e43c31a8adccc1d301bed26d8f/addon/record-data.js#L56) and  [`setupFragmentArray`](https://github.com/adopted-ember-addons/ember-data-model-fragments/blob/1403665eb8b8a3e43c31a8adccc1d301bed26d8f/addon/record-data.js#L187), which are what are used when bootstrapping fragments. This works correctly and the `willCommit` hooks fire as expected and the fragments get their values updated from the server responses.

A problem I'm now seeing as a result of this change is that  `this.fragments` is used to calculate `changedAttributes`. My spike then causes all fragments to appear in `changedAttributes()` even just after a model with fragments has just been initialized. So I think a deeper refactor here would need to be figured out if we wanted `willCommit` to be called properly.

@knownasilya I'm going to test this specific fix in my app now to see whether it fixes our problem. If it does it would be a good fix to unblock an upgrade for us

Reply:
I spiked out a deeper refactor in #417

Reply:
Nice work @dwickern. I'll take a closer look at that PR when I get a chance at some point this week